pipeline_api/src/main/java/com/splicemachine/pipeline/writehandler/UpdateUtils.java

```diff
@@ -0,0 +1,84 @@
+import com.splicemachine.storage.index.BitIndex;
+import com.splicemachine.storage.index.BitIndexing;
+import com.splicemachine.utils.ByteSlice;
+
+public class UpdateUtils {
+
+    public static KVPair deleteFromUpdate(KVPair mutation) {
+        EntryDecoder rowDecoder = new EntryDecoder();
+        rowDecoder.set(mutation.getValue());
+        BitIndex index = rowDecoder.getCurrentIndex();
+        MultiFieldDecoder fieldDecoder = rowDecoder.getEntryDecoder();
+        int cardinality = index.cardinality();
+        for(int i=index.nextSetBit(0), c=0; c<cardinality/2; i=index.nextSetBit(i+1), c++){
+            rowDecoder.seekForward(fieldDecoder, i);
+        }
+        byte[] data = fieldDecoder.slice(fieldDecoder.length() - fieldDecoder.offset());
+        BitIndex resultIndex = BitIndexing.getBestIndex(
+                halveSet(index.getFields()),
+                halveSet(index.getScalarFields()),
+                halveSet(index.getFloatFields()),
+                halveSet(index.getDoubleFields()));
+        byte[] bitData = resultIndex.encode();
+        byte[] result = new byte[bitData.length+data.length+1];
+        System.arraycopy(bitData, 0, result, 0, bitData.length);
+        result[bitData.length] = 0;
+        System.arraycopy(data,0,result,bitData.length+1,data.length);
+
+        ByteSlice rowSlice = mutation.rowKeySlice();
+        return new KVPair(
+                rowSlice.array(),rowSlice.offset(),rowSlice.length(),
+                result,0,result.length,KVPair.Type.DELETE);
+    }
+
+    public static KVPair getBaseUpdateMutation(KVPair mutation) {
+        EntryDecoder rowDecoder = new EntryDecoder();
+        rowDecoder.set(mutation.getValue());
+        BitIndex index = rowDecoder.getCurrentIndex();
+        MultiFieldDecoder fieldDecoder = rowDecoder.getEntryDecoder();
+        int offset = fieldDecoder.offset();
+        for(int i=index.nextSetBit(0), c=0; c<index.cardinality()/2; i=index.nextSetBit(i+1), c++){
+            rowDecoder.seekForward(fieldDecoder, i);
+        }
+        int end = fieldDecoder.offset();
+        fieldDecoder.seek(offset);
+        byte[] data = fieldDecoder.slice(end - offset - 1); // -1 to remove the NULL separator
+        BitIndex resultIndex = BitIndexing.getBestIndex(
+                halveSet(index.getFields()),
+                halveSet(index.getScalarFields()),
+                halveSet(index.getFloatFields()),
+                halveSet(index.getDoubleFields()));
+        byte[] bitData = resultIndex.encode();
+        byte[] result = new byte[bitData.length+data.length+1];
+        System.arraycopy(bitData, 0, result, 0, bitData.length);
+        result[bitData.length] = 0;
+        System.arraycopy(data,0,result,bitData.length+1,data.length);
+
+        ByteSlice rowSlice = mutation.rowKeySlice();
+        return new KVPair(
+                rowSlice.array(),rowSlice.offset(),rowSlice.length(),
+                result,0,result.length,KVPair.Type.UPDATE);
```

Comment:
except for line 23/52 and KVPair.Type.UPDATE and KVPair.Type.DELETE, functions
deleteFromUpdate and getBaseUpdateMutation are the same.
Please unify.

Reply:
You are right, consolidated in single method. Thanks for the review!